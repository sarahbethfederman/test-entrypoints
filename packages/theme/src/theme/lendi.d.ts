export declare const theme: {
  container: {
    width: string;
  };
  breakpoints: {
    mobile: number;
    tablet: string;
    desktop: string;
  };
  spacing: {
    0: number;
    1: string;
    2: string;
    3: string;
    4: string;
    5: string;
    6: string;
  };
  typography: {
    heading: {
      fontFamily: string;
      sizes: {
        fontSize: string;
        lineHeight: number;
        fontWeight: string;
      }[];
    };
    body: {
      fontFamily: string;
      sizes: {
        fontSize: string;
        lineHeight: number;
      }[];
    };
  };
  colors: {
    brand: {
      primary: string;
      'primary.dark': string;
      'primary.medium': string;
      'primary.light': string;
      secondary: string;
      'secondary.dark': string;
      'secondary.medium': string;
      'secondary.light': string;
      tertiary: string;
      'tertiary.dark': string;
      'tertiary.medium': string;
      'tertiary.light': string;
      quaternary: string;
    };
    shade: string[];
    alert: {
      info: string;
      'info.dark': string;
      'info.light': string;
      warn: string;
      'warn.dark': string;
      'warn.light': string;
      error: string;
      'error.dark': string;
      'error.light': string;
      success: string;
      'success.dark': string;
      'success.light': string;
    };
  };
};
