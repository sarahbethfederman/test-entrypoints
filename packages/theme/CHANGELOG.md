## [9.1.0] 2020-06-29 [FUNNEL-1473](https://creditandfinance.atlassian.net/browse/FUNNEL-1473)

## 10.0.0-beta.0

### Minor Changes

- preconstruct infra update

### Patch Changes

- Updated dependencies [undefined]
  - @lendi-ui/breakpoint@6.0.0-beta.0
- Change `--lend-ui-size` to use new t-shirt breakpoints.

## [9.0.0] 2020-06-29

### Changed

- lendi-ui-size root size breakpoint changed to 576px instead of 577px

## [8.1.0] 2019-10-02 [HUB-589](https://creditandfinance.atlassian.net/browse/HUB-589)

### Changed

- Enhancement LUI theme with default font-family for body and font-weight for heading.

## [8.0.1] 2019-9-29 [HUB-589](https://creditandfinance.atlassian.net/browse/HUB-589)

### Changed

- added font family defaults based on theme

## [8.0.0] 2019-9-23 - 2019-08-15 - [HUB-412](https://creditandfinance.atlassian.net/browse/HUB-412)

## [7.0.0] 2019-9-23 - 2019-08-15 - [HUB-412](https://creditandfinance.atlassian.net/browse/HUB-412)

### Changed

- Color is now a type
- ColorPrefix is now a type
- theme.color object has been flattened
- Update typescript to version 3.5.2 and react to version 16.8.4

## [6.2.0] - 2019-05-17 - [HUB-174](https://creditandfinance.atlassian.net/browse/HUB-174)

### Changed

- Module format reverted back to CJS.

### Deleted

- Dropped "module" property from package.json.

## [6.1.1] - 2019-05-10 - [HUB-137](https://creditandfinance.atlassian.net/browse/HUB-137)

### Added

- Add `-ms-clear` into theme injectGlobal to prevent cross icon in input by Edge and IE browser default style

## [6.1.0] - 2019-05-02 - [HUB-88](https://creditandfinance.atlassian.net/browse/HUB-88)

### Changed

- replace `border-radius: 6px` with select function in theme

## [6.0.2] - 2019-05-09 - [HUB-92](https://creditandfinance.atlassian.net/browse/HUB-92)

### Changed

- Updated CHANGELOG to match required parsing format.

## [6.0.1] - 2019-05-08 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Changed

- Change module main entry point to esm in package.json

## [6.0.0] - 2019-04-30 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Deleted

- Remove the CJS bundles.

## [5.0.0] - 2019-03-06 - [BRAVO-387](https://creditandfinance.atlassian.net/browse/BRAVO-387)

### Changed

- Added optional theme kind prop.

## [4.0.2] - 2019-02-25 - [BRAVO-511](https://creditandfinance.atlassian.net/browse/BRAVO-511)

### Changed

- Change warning colour to meet accessibility standards.

## [4.0.1]

### Added

- Added CHANGELOG file.
