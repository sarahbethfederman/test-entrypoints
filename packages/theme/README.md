# @lendi-ui/theme

Lendi theme provider.

## Installation

```
yarn add @lendi-ui/theme
```

## Usage

```jsx
import Theme from '@lendi-ui/theme';
<Theme>
  <App />
</Theme>;
```
