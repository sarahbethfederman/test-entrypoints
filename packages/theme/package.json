{
  "name": "@lendi-ui/theme",
  "version": "9.1.0",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/breakpoint": "^5.0.0",
    "@types/react": "^16.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.8",
    "styled-components": "^4"
  },
  "scripts": {
    "build:cjs": "tsc --project tsconfig.cjs.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:cjs"
  }
}
