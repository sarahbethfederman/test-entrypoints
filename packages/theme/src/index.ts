export * from './select';
export { theme } from './theme/lendi';
export { Theme as default } from './Theme';
export { ThemeMap, Colors, ColorsMap, ColorPrefixes } from './types';
