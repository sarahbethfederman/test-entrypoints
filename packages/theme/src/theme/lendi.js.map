{"version":3,"file":"lendi.js","sourceRoot":"","sources":["lendi.ts"],"names":[],"mappings":";;AAAA,IAAM,SAAS,GAAG;IAChB,KAAK,EAAE,QAAQ;CAChB,CAAC;AAEF,IAAM,WAAW,GAAG;IAClB,MAAM,EAAE,CAAC;IACT,MAAM,EAAE,WAAW;IACnB,OAAO,EAAE,WAAW;CACrB,CAAC;AAEF,IAAM,OAAO,GAAG;IACd,CAAC,EAAE,CAAC;IACJ,CAAC,EAAE,KAAK;IACR,CAAC,EAAE,KAAK;IACR,CAAC,EAAE,MAAM;IACT,CAAC,EAAE,MAAM;IACT,CAAC,EAAE,MAAM;IACT,CAAC,EAAE,MAAM;CACV,CAAC;AAEF,IAAM,UAAU,GAAG;IACjB,OAAO,EAAE;QACP,UAAU,EAAE,qBAAqB;QACjC,KAAK,EAAE;YACL;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;gBACnB,UAAU,EAAE,MAAM;aACnB;YACD;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;gBACnB,UAAU,EAAE,WAAW;aACxB;YACD;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;gBACnB,UAAU,EAAE,WAAW;aACxB;YACD;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;gBACnB,UAAU,EAAE,WAAW;aACxB;YACD;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;gBACnB,UAAU,EAAE,WAAW;aACxB;YACD;gBACE,QAAQ,EAAE,IAAI;gBACd,UAAU,EAAE,EAAE,GAAG,EAAE;gBACnB,UAAU,EAAE,WAAW;aACxB;SACF;KACF;IAED,IAAI,EAAE;QACJ,UAAU,EAAE,yBAAyB;QACrC,KAAK,EAAE;YACL;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;aACpB;YACD;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;aACpB;YACD;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;aACpB;YACD;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;aACpB;YACD;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;aACpB;YACD;gBACE,QAAQ,EAAE,MAAM;gBAChB,UAAU,EAAE,EAAE,GAAG,EAAE;aACpB;SACF;KACF;CACF,CAAC;AAEF,IAAM,MAAM,GAAG;IACb,KAAK,EAAE;QACL,OAAO,EAAE,SAAS;QAClB,cAAc,EAAE,SAAS;QACzB,gBAAgB,EAAE,SAAS;QAC3B,eAAe,EAAE,SAAS;QAE1B,SAAS,EAAE,SAAS;QACpB,gBAAgB,EAAE,SAAS;QAC3B,kBAAkB,EAAE,SAAS;QAC7B,iBAAiB,EAAE,SAAS;QAE5B,QAAQ,EAAE,SAAS;QACnB,eAAe,EAAE,SAAS;QAC1B,iBAAiB,EAAE,SAAS;QAC5B,gBAAgB,EAAE,SAAS;QAE3B,UAAU,EAAE,SAAS;KACtB;IACD,KAAK,EAAE,CAAC,SAAS,EAAE,SAAS,EAAE,SAAS,EAAE,SAAS,EAAE,SAAS,EAAE,SAAS,EAAE,SAAS,EAAE,SAAS,EAAE,SAAS,CAAC,CAAC,OAAO,EAAE;IACpH,KAAK,EAAE;QACL,IAAI,EAAE,SAAS;QACf,WAAW,EAAE,SAAS;QACtB,YAAY,EAAE,SAAS;QAEvB,IAAI,EAAE,SAAS;QACf,WAAW,EAAE,SAAS;QACtB,YAAY,EAAE,SAAS;QAEvB,KAAK,EAAE,SAAS;QAChB,YAAY,EAAE,SAAS;QACvB,aAAa,EAAE,SAAS;QAExB,OAAO,EAAE,SAAS;QAClB,cAAc,EAAE,SAAS;QACzB,eAAe,EAAE,SAAS;KAC3B;CACF,CAAC;AAEW,QAAA,KAAK,GAAG;IACnB,SAAS,WAAA;IACT,WAAW,aAAA;IACX,OAAO,SAAA;IACP,UAAU,YAAA;IACV,MAAM,QAAA;CACP,CAAC"}