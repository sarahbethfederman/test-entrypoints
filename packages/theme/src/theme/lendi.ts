const typography = {
  heading: {
    fontFamily: "'Cabin', sans-serif",
  },

  body: {
    fontFamily: "'Open Sans', sans-serif",
  },
};

const colors = {
  primary: {
    '50': '#e5f8f6',
    '100': '#ccf2ed',
    '200': '#99e5db',
    '300': '#66d9c9',
    '400': '#32ccb7',
    '500': '#00c0a5',
    '600': '#009984',
    '700': '#007363',
    '800': '#004c42',
    '900': '#002621',
    '950': '#001310',
  },
  secondary: {
    '50': '#e6eef0',
    '100': '#cedee1',
    '200': '#9ebdc4',
    '300': '#6d9ca7',
    '400': '#3d7a8a',
    '500': '#0d5a6d',
    '600': '#0a4857',
    '700': '#073641',
    '800': '#05242b',
    '900': '#021215',
    '950': '#01080a',
  },
  tertiary: {
    '50': '#fff0ef',
    '100': '#ffe2df',
    '200': '#ffc5c0',
    '300': '#ffa9a1',
    '400': '#ff8c82',
    '500': '#ff7063',
    '600': '#cc594f',
    '700': '#99433b',
    '800': '#662c27',
    '900': '#331613',
    '950': '#190b09',
  },
  shade: {
    '0': '#ffffff',
    '50': '#f1f1f1',
    '100': '#e3e3e3',
    '200': '#c8c8c8',
    '300': '#adadad',
    '400': '#929292',
    '500': '#777777',
    '600': '#5f5f5f',
    '700': '#474747',
    '800': '#2f2f2f',
    '900': '#171717',
    '950': '#0b0b0b',
    '1000': '#000000',
  },
  info: {
    '50': '#ecf3fb',
    '100': '#dae7f7',
    '200': '#b5d0ef',
    '300': '#91b9e8',
    '400': '#6ca2e0',
    '500': '#488bd9',
    '600': '#396fad',
    '700': '#2b5382',
    '800': '#1c3756',
    '900': '#0e1b2b',
    '950': '#070d15',
  },
  warn: {
    '50': '#fef6e9',
    '100': '#fdedd3',
    '200': '#fbdba7',
    '300': '#f9c97b',
    '400': '#f7b74e',
    '500': '#f5a623',
    '600': '#c4841c',
    '700': '#936315',
    '800': '#62420e',
    '900': '#312107',
    '950': '#181003',
  },
  error: {
    '50': '#ffe8ec',
    '100': '#ffd1da',
    '200': '#ffa4b6',
    '300': '#ff7692',
    '400': '#ff496e',
    '500': '#ff1c4a',
    '600': '#cc163b',
    '700': '#99102c',
    '800': '#660b1d',
    '900': '#33050e',
    '950': '#190207',
  },
  success: {
    '50': '#edf8e5',
    '100': '#dbf2cc',
    '200': '#b8e599',
    '300': '#94d966',
    '400': '#71cc32',
    '500': '#4ec000',
    '600': '#3e9900',
    '700': '#2e7300',
    '800': '#1f4c00',
    '900': '#0f2600',
    '950': '#071300',
  },
};

export const theme = {
  typography,
  colors,
};
