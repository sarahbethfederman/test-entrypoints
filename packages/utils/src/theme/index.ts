export type ColorKeys = '25' | '50' | '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900' | '950';

export type ColorsMap = { [key in ColorKeys]: string };

export type Typography = {
  heading: {
    fontFamily: string;
  };

  body: {
    fontFamily: string;
  };
};

export type Logo = {
  logoName: string;
};

export type Colors = {
  primary: ColorsMap;
  secondary: ColorsMap;
  tertiary: ColorsMap;
  shade: ColorsMap;
  info: ColorsMap;
  warn: ColorsMap;
  error: ColorsMap;
  success: ColorsMap;
};

export type ThemeMap = {
  typography: Typography;
  colors: Colors;
  logo: Logo;
};
