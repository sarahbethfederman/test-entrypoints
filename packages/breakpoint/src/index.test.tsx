import 'jest-styled-components';
import * as React from 'react';
import styled, { css, SimpleInterpolation } from 'styled-components';
import { mount } from 'enzyme';
import { Colors } from '@lendi-ui/theme';
import { gte, between, map, Breakpoint, BreakpointValue, BreakpointValueMap, getBreakpoint } from '.';
import { match } from 'css-mediaquery';

interface ComplexComponentProps {
  color: Colors;
}

describe('gte()', () => {
  it('should wrap styles in a media rule', () => {
    const Component = styled.div`
      ${gte('mobile')`
        color: red;
      `}
      ${gte('tablet')`
        color: green;
      `}
      ${gte('desktop')`
        color: blue;
      `}
    `;

    const element = mount(<Component />);
    expect(element).toHaveStyleRule('color', 'red', {
      media: `(min-width:${Breakpoint.mobile})`,
    });
    expect(element).toHaveStyleRule('color', 'green', {
      media: `(min-width:${Breakpoint.tablet})`,
    });
    expect(element).toHaveStyleRule('color', 'blue', {
      media: `(min-width:${Breakpoint.desktop})`,
    });
  });

  it('should wrap styles consisting of functions in a media rule', () => {
    const ComplexComponent = styled.div<ComplexComponentProps>`
      ${gte('tablet')`
        color: ${((({ color }) => color) as unknown) as SimpleInterpolation};
      `};
    `;

    const element = mount(<ComplexComponent color="primary.100" />);
    expect(element).toHaveStyleRule('color', 'primary.100', {
      media: `(min-width:${Breakpoint.tablet})`,
    });
  });
});

describe('between()', () => {
  it('should wrap styles in a media rule', () => {
    const Component = styled.div`
      ${between('mobile', 'tablet')`
        color: red;
      `};
      ${between('tablet', 'desktop')`
        color: green;
      `}
      color: blue;
    `;

    const element = mount(<Component />);
    expect(element).toHaveStyleRule('color', 'red', {
      media: `(min-width:${Breakpoint.mobile}) and (max-width:${Breakpoint.tablet})`,
    });
    expect(element).toHaveStyleRule('color', 'green', {
      media: `(min-width:${Breakpoint.tablet}) and (max-width:${Breakpoint.desktop})`,
    });
    expect(element).toHaveStyleRule('color', 'blue');
  });

  it('should wrap styles consisting of functions in a media rule', () => {
    const ComplexComponent = styled.div`
      ${between('tablet', 'desktop')`
        color: ${(((props: { color: Colors }) => props.color) as unknown) as SimpleInterpolation};
      `};
    `;

    const element = mount(<ComplexComponent color="purple" />);
    expect(element).toHaveStyleRule('color', 'purple', {
      media: `(min-width:${Breakpoint.tablet}) and (max-width:${Breakpoint.desktop})`,
    });
  });
});

describe('map()', () => {
  interface SimpleMapComponentProps {
    bg: BreakpointValue<string> | BreakpointValueMap<string>;
  }

  const SimpleMapComponent = styled.div`
    ${(props: SimpleMapComponentProps) => map(props.bg, (val = 'unset') => `background-color: ${val};`)};
  `;

  interface ComplexMapComponentProps {
    size: BreakpointValue<number> | BreakpointValueMap<number>;
    multiplier: number;
  }

  const ComplexMapComponent = styled.div`
    ${(props: ComplexMapComponentProps) =>
      map(
        props.size,
        (val = 1) =>
          css`
            width: ${(childProps: ComplexMapComponentProps) => childProps.multiplier * val * 100}%;
          `
      )};
  `;

  interface ResponsiveMapComponentProps {
    size: BreakpointValue<string> | BreakpointValueMap<string>;
  }

  const ResponsiveMapComponent = styled.div`
    ${(props: ResponsiveMapComponentProps) => map(props.size, (val) => `width:${val}`)};
  `;

  it('should not wrap styles in a media rule', () => {
    const element = mount(<SimpleMapComponent bg="red" />);
    expect(element).toHaveStyleRule('background-color', 'red');
  });

  it('should wrap styles in a media rule at each breakpoint', () => {
    const element = mount(<SimpleMapComponent bg={{ mobile: 'red', tablet: 'green', desktop: 'blue' }} />);
    expect(element).toHaveStyleRule('background-color', 'unset');
    expect(element).toHaveStyleRule('background-color', 'red', {
      media: `(min-width:${Breakpoint.mobile})`,
    });
    expect(element).toHaveStyleRule('background-color', 'green', {
      media: `(min-width:${Breakpoint.tablet})`,
    });
    expect(element).toHaveStyleRule('background-color', 'blue', {
      media: `(min-width:${Breakpoint.desktop})`,
    });
  });

  it('should wrap styles in a media rule at each tshirt breakpoint', () => {
    const element = mount(
      <SimpleMapComponent bg={{ xs: 'red', sm: 'green', md: 'blue', lg: 'yellow', xl: 'orange' }} />
    );
    expect(element).toHaveStyleRule('background-color', 'unset');
    expect(element).toHaveStyleRule('background-color', 'red', {
      media: `(min-width:${Breakpoint.xs})`,
    });
    expect(element).toHaveStyleRule('background-color', 'green', {
      media: `(min-width:${Breakpoint.sm})`,
    });
    expect(element).toHaveStyleRule('background-color', 'blue', {
      media: `(min-width:${Breakpoint.md})`,
    });
    expect(element).toHaveStyleRule('background-color', 'yellow', {
      media: `(min-width:${Breakpoint.lg})`,
    });
    expect(element).toHaveStyleRule('background-color', 'orange', {
      media: `(min-width:${Breakpoint.xl})`,
    });
  });

  it('should throw an error if mixing descriptive and tshirt breakpoints', () => {
    expect(() => mount(<SimpleMapComponent bg={{ xs: 'red', mobile: 'blue' }} />)).toThrow(
      new Error('Mobile and xs are synonyms for the same breakpoint, please use only one.')
    );
  });

  it('should not wrap styles consisting of functions in a media rule', () => {
    const element = mount(<ComplexMapComponent size={1 / 2} multiplier={2} />);
    expect(element).toHaveStyleRule('width', '100%');
  });

  it('should wrap styles consisting of functions in a media rule at each breakpoint', () => {
    const element = mount(<ComplexMapComponent size={{ mobile: 1, tablet: 1 / 2, desktop: 1 / 4 }} multiplier={2} />);
    expect(element).toHaveStyleRule('width', '200%');
    expect(element).toHaveStyleRule('width', '200%', {
      media: `(min-width:${Breakpoint.mobile})`,
    });
    expect(element).toHaveStyleRule('width', '100%', {
      media: `(min-width:${Breakpoint.tablet})`,
    });
    expect(element).toHaveStyleRule('width', '50%', {
      media: `(min-width:${Breakpoint.desktop})`,
    });
  });

  it('reponsiveness with different order of screen size', () => {
    let element = mount(<ResponsiveMapComponent size={{ tablet: '50%', desktop: '100%', mobile: '25%' }} />);
    expect(element).toHaveStyleRule('width', '25%', {
      media: `(min-width:${Breakpoint.mobile})`,
    });
    expect(element).toHaveStyleRule('width', '50%', {
      media: `(min-width:${Breakpoint.tablet})`,
    });
    expect(element).toHaveStyleRule('width', '100%', {
      media: `(min-width:${Breakpoint.desktop})`,
    });
    expect(element).toMatchSnapshot();
    element = mount(<ResponsiveMapComponent size={{ tablet: '50%', mobile: '25%', desktop: '100%' }} />); // change the order and check the snapshot
    expect(element).toMatchSnapshot();
  });
});

describe('getBreakpoint', () => {
  it.each([
    [1201, 'desktop'],
    [577, 'tablet'],
    [360, 'mobile'],
  ])('when window width is %s, should return "%s"', (width: number, result: string) => {
    window.matchMedia = jest.fn().mockImplementation((query: string) => ({
      matches: match(query, { type: 'screen', width: `${width}px` }),
    }));

    expect(getBreakpoint()).toBe(result);
  });
});
