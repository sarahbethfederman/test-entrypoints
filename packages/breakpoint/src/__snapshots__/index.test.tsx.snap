// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map() reponsiveness with different order of screen size 1`] = `
.c0 {
  width: undefined;
}

@media (min-width:0) {
  .c0 {
    width: 25%;
  }
}

@media (min-width:36.0625em) {
  .c0 {
    width: 50%;
  }
}

@media (min-width:75.0625em) {
  .c0 {
    width: 100%;
  }
}

<styled.div
  size={
    Object {
      "desktop": "100%",
      "mobile": "25%",
      "tablet": "50%",
    }
  }
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
    ",
            [Function],
            ";
  ",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "usesTheme": false,
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    size={
      Object {
        "desktop": "100%",
        "mobile": "25%",
        "tablet": "50%",
      }
    }
  >
    <div
      className="c0"
      size={
        Object {
          "desktop": "100%",
          "mobile": "25%",
          "tablet": "50%",
        }
      }
    />
  </StyledComponent>
</styled.div>
`;

exports[`map() reponsiveness with different order of screen size 2`] = `
.c0 {
  width: undefined;
}

@media (min-width:0) {
  .c0 {
    width: 25%;
  }
}

@media (min-width:36.0625em) {
  .c0 {
    width: 50%;
  }
}

@media (min-width:75.0625em) {
  .c0 {
    width: 100%;
  }
}

<styled.div
  size={
    Object {
      "desktop": "100%",
      "mobile": "25%",
      "tablet": "50%",
    }
  }
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
    ",
            [Function],
            ";
  ",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "usesTheme": false,
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    size={
      Object {
        "desktop": "100%",
        "mobile": "25%",
        "tablet": "50%",
      }
    }
  >
    <div
      className="c0"
      size={
        Object {
          "desktop": "100%",
          "mobile": "25%",
          "tablet": "50%",
        }
      }
    />
  </StyledComponent>
</styled.div>
`;
